validator-api/validator_api/services/scoring_service.py

```diff
@@ -266,19 +271,52 @@ async def get_detailed_video_description_embedding_score(self, video_id, task_ov
             return detailed_video_description, None, None
         return detailed_video_description, embedding, await self.get_description_uniqueness_score(embedding)
 
+    async def get_boosted_multiplier(self, focusing_task: str, focusing_description: str) -> float:
+        """
+        Get boosted tasks from the database "boosted_tasks" table
+        ask Gemini if the task matches any of the boosted tasks
+        return the multiplier for the task if it is boosted, otherwise 1.0
+        """
+        with get_db_context() as db:
+            boosted_tasks = db.query(BoostedTask).all()
+            system_prompt = focus_scoring_prompts.BOOST_SCORING_SYSTEM_PROMPT.format(boosted_tasks="\n".join([f"{idx}. {task.title}: {task.description}" for idx, task in enumerate(boosted_tasks)]))
+            user_prompt = focus_scoring_prompts.BOOST_SCORING_USER_PROMPT.format(
+                focusing_task=focusing_task,
+                focusing_description=focusing_description,
+            )
+            boosted_task_index = await self.make_gemini_request_with_retries(
+                system_prompt=system_prompt,
+                user_prompt=user_prompt,
+                video_id=None,
+                OutputClassSchema=BoostedTaskIndex,
+            )
+            # print(f"Boosted task index: {boosted_task_index.index}")
+            if boosted_task_index.index == -1 or boosted_task_index.index >= len(boosted_tasks):
+                return 1.0
+            multiplier = boosted_tasks[boosted_task_index.index].multiplier
+            try:
+                multiplier = float(multiplier)
+            except (TypeError, ValueError):
+                print(f"Invalid task score boost multiplier: {multiplier}, returning 1.0")
+                multiplier = 1.0
+            return multiplier
```

Comment:
I believe you can free the DB session before making the gemini request. best to not hog DB sessions for longer than necessary. lmk if i am incorrect here, but it seems like you don't need any further data from the DB after just getting the boosted tasks list. Though you would need to make sure to get the boosted tasks list as pydantic objects rather than database objects since you would need to refer to that data later and the DB connection would be closed by then. See this as an example: https://github.com/omegalabsinc/omegalabs-bittensor-subnet/blob/6f8e69aefa9e6a3291e2291d0cc76b4710419c6a/validator-api/validator_api/database/crud/focusvideo.py#L307-L310